mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/AnnotatedTypeSource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.definition.annotation;
```

Comment:
In theory it could be used for ORM too. I was contemplating just moving all such config over from `AnnotationMappingConfigurationContext` to `AnnotatedTypeSource`, but yeah, we'll have another look when we deal with HSEARCH-5076.

In any case, I marked this class as `@Incubating`, like everything in the Standalone Pojo Mapper, so that we're free to move it around.